### PR TITLE
DOC: Remove flake8 errors for basics.rst and contributing_docstring.rst

### PR DESCRIPTION
**Pull request by saurav2608**
xref: #24173

### CODE REVIEW

**Comment by jreback** [comment]
need to merge master

File: doc/source/basics.rst [hunk 1]
```diff
@@ -807,15 +807,15 @@ Compare the following
 .. code-block:: python
 
    # f, g, and h are functions taking and returning ``DataFrames``
-   >>> f(g(h(df), arg1=1), arg2=2, arg3=3)
+   >>> f(g(h(df), arg1=1), arg2=2, arg3=3)  # noqa F821
```
Comment (jreback) [inline doc/source/basics.rst]: hmm, this is really ugly, @datapythonista ?

**Comment by datapythonista** [comment]
Sorry, I wasn't available the last couple of days. That's the reason I didn't create the issues for `basics` and `contributing_docstring`, because they are not so simple. Can you just leave the easy changes (all the ones without the `noqa` (adding `np`...) here? I'll take care of the remaining myself in a separate PR later.

**Comment by jreback** [comment]
thanks @saurav2608

**Comment by iamshreeram** [comment]
@saurav2608  / @jreback , I still see `flake8-rst` issues with `contributing_docstring.rst`. Is it good idea to add `# noqa` for below issues ?

```
contributing_docstring.rst:52:9: F821 undefined name 'add'
contributing_docstring.rst:54:9: F821 undefined name 'add'
contributing_docstring.rst:56:9: F821 undefined name 'add'
contributing_docstring.rst:456:16: F821 undefined name 'random'
contributing_docstring.rst:476:18: F821 undefined name 'random'
contributing_docstring.rst:477:27: F821 undefined name 'random'
contributing_docstring.rst:477:41: F821 undefined name 'string'
contributing_docstring.rst:498:19: F821 undefined name 'random'
contributing_docstring.rst:650:21: F821 undefined name 'pd'
contributing_docstring.rst:774:60: F821 undefined name 'numpy'
contributing_docstring.rst:832:31: F821 undefined name 'numpy'
contributing_docstring.rst:950:9: F821 undefined name 'Substitution'
contributing_docstring.rst:951:9: F821 undefined name 'Appender'
contributing_docstring.rst:956:9: F821 undefined name 'Substitution'
contributing_docstring.rst:957:9: F821 undefined name 'Appender'
contributing_docstring.rst:986:5: F821 undefined name 'Appender'
contributing_docstring.rst:986:14: F821 undefined name 'template'
contributing_docstring.rst:986:25: F821 undefined name '_shared_doc_kwargs'
```

**Comment by datapythonista** [comment]
Thanks for offering @iamshreeram, but we may be moving that page our of the docs into the website, so not worth fixing those at this point. You can easily find things to fix in the docstrings, those will be much more useful. `./scripts/validate_docstrings.py` should give you lots of errors to fix. :)

**Comment by steveayers124** [comment]
We're hunting errors to fix, and I'm just getting started, so we appreciate your pointers, @datapythonista !

**Comment by steveayers124** [comment]
> ./scripts/validate_docstrings.py
@datapythonista , I'm seeing a lot to sift through to find something I can fix. What advice do you have to point me in the right direction?

**Comment by datapythonista** [comment]
are you in the Walmart hackathon? how people is communicating there, do you have any IM?

**Comment by iamshreeram** [comment]
@datapythonista , Yes. https://pandashack.slack.com/ :)

**Comment by datapythonista** [comment]
Do you think I can get an invitation?

**Comment by iamshreeram** [comment]
Checking.. Just a moment

**Comment by iamshreeram** [comment]
Please try - http://bit.ly/pandashackslack

**Comment by steveayers124** [comment]
@datapythonista , are you able to join?

**Pull request merged**